src/utils/ao.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { connect } from '@permaweb/aoconnect';
+
+import ANT_LUA from '../ant-lua-code.js';
+import { AOS_MODULE_ID, DEFAULT_SCHEDULER_ID } from '../constants.js';
+import { ANTState } from '../contract-state.js';
+import { AoClient, Logger } from '../types.js';
+
+/**
+ * Spawns an aos process and loads the ANT code. Optionally initializes state.
+ * @param param0 configuration for the spawn and eval process.
+ * @param options - logger, tags, data - optional, applied to the spawn message.
+ * @returns @type {Promise<string | undefined>} - processId or undefined
+ *
+ * @example
+ * Basic usage:
+ * ```typescript
+ * import { createDataItemSigner } from '@permaweb/aoconnect';
+ * const antId = await spawnANT({
+ *  signer: createDataItemSigner(window.arweaveWallet)
+ * })
+ * ```
+ */
+export async function spawnANT(
+  {
+    module = AOS_MODULE_ID,
+    luaString = ANT_LUA,
+    ao = connect(),
+    scheduler = DEFAULT_SCHEDULER_ID,
+    signer,
+    state,
```

Comment:
add the txId of the source code for the bundled lua code and default it to our current tx. then store it locally on the filesystem in `tmp/arns/ant.lua` file

Reply:
added a todo for this.